libs/isograph-react/src/core/cache.ts

```diff
@@ -335,15 +352,29 @@ function callSubscriptions(
   );
 }
 
-function hasOverlappingIds(set1: Set<DataId>, set2: Set<DataId>): boolean {
-  for (const id of set1) {
-    if (set2.has(id)) {
-      return true;
+function hasOverlappingIds(
+  set1: EncounteredIds,
+  set2: EncounteredIds,
+): boolean {
+  for (const typeName in set1) {
+    const set2Ids = set2[typeName];
+    if (!set2Ids) {
+      continue;
+    }
+
+    for (const id of set1[typeName]) {
```

Comment:
this will likely perform better if we assign `set1[typeName]` to an intermediate variable and avoid the lookup on every iteration.

Maybe this can be simplified with https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set/isDisjointFrom. (I'll add a task to figure out polyfills)

Reply:
#179

Reply:
This should be good, both `set1[typeName]` and `set2[typeName]`, happens once per iteration of the outer loop